### DISCUSSION TITLE
Section 5.24 Site Estático

### DISCUSSION THREAD

**Discussion by eduardocortez3** [chgasparoto/curso-aws-com-terraform#100] [Q&A]
Prof. Cleber.
Caso eu queira reaproveitar o código para criar um novo ambiente, por exemplo, quero criar um ambiente de sandbox e outro de desenvolvimento, o que eu preciso fazer?
Eu imagino que, copiamos todos os arquivos *.tf, *.tfvars e *.hcl de um diretório de origem, por exemplo /producao-app para o diretório do novo ambiente /desenvol-app, estando no novo diretório rodamos o comando terraform init, claro, após ajustar o código, principalmente o backend do tfstate.
Seria isso?

**Comment by chgasparoto** [comment]
@eduardocortez3 você não precisa copiar todos os arquivos, apenas os **.tfvars** e o **backend**. 

Eu já utilizei duas maneiras para separar os ambientes.

1. Utilizando workspaces
2. Criando pastas específicas para cada ambiente, aqui nesse [repositório](https://github.com/chgasparoto/youtube-cleber-gasparoto/tree/main/0003-terraform-gha) tem um exemplo. E aqui um vídeo meu no [youtube](https://youtu.be/2McC52KuWh0).
